rules/DeadCode/Rector/StmtsAwareInterface/RemoveJustPropertyFetchRector.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DeadCode\Rector\StmtsAwareInterface;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Stmt\Expression;
+use PHPStan\PhpDocParser\Ast\PhpDoc\VarTagValueNode;
+use Rector\Core\Contract\PhpParser\Node\StmtsAwareInterface;
+use Rector\Core\Rector\AbstractRector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\ReadWrite\NodeFinder\NodeUsageFinder;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\DeadCode\Rector\StmtsAwareInterface\RemoveJustPropertyFetchRector\RemoveJustPropertyFetchRectorTest
+ */
+final class RemoveJustPropertyFetchRector extends AbstractRector
+{
+    public function __construct(
+        private NodeUsageFinder $nodeUsageFinder
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Inline property fetch assign to a variable, that has no added value', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    private $name;
+
+    public function run()
+    {
+        $name = $this->name;
+
+        return $name;
+    }
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    private $name;
+
+    public function run()
+    {
+        return $this->name;
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [StmtsAwareInterface::class];
+    }
+
+    /**
+     * @param StmtsAwareInterface $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $variableUsages = [];
+        $propertyFetch = null;
+
+        foreach ((array) $node->stmts as $key => $stmt) {
+            if (! $stmt instanceof Expression) {
+                continue;
+            }
+
+            if (! $stmt->expr instanceof Assign) {
+                continue;
+            }
+
+            // assign to a variable
+            // ...
+            $assign = $stmt->expr;
+
+            if (! $assign->expr instanceof PropertyFetch) {
```

Comment:
how about static property fetch ?

Reply:
In case of static property fetch, some reading/writing/cache is expected, so it would be risky to try to cover it.

This case should cover readonly property fetch assign.

Reply:
Btw, seems like `--debug`/`-vvv` option does not work anymore. Only the last error message is printed, not the full stack trace. I tired to downgrade to Symfony 6 but it was not helpful :/ 

Could you check it? :slightly_smiling_face:

Reply:
I will try. How to reproduce the not shown stack trace for your use case? throw Exception?

Reply:
Yea, I just put an exception to e.g. `PhpDocInfoFactory::createFromNodeOrEmpty()` on first line.